libs/ddd-angular/src/helpers/domain-helper.ts

```diff
@@ -1,19 +1,97 @@
 import { Tree, readProjectConfiguration } from '@nrwl/devkit';
+import { DDDObject } from '../model/ddd.model';
+import { Domain } from '../model/domain.model';
+import { API } from '../model/api.model';
+import { Shell } from '../model/shell.model';
+import { Feature } from '../model/feature.model';
+import { UI } from '../model/ui.model';
+import { Util } from '../model/util.model';
 
+/**
+ * It throws an error if the domain is invalid or shared
+ *
+ * @param tree The virtual filesystem tree provided by NX
+ * @param domain The domain name
+ */
 export const guardValidDomain = (tree: Tree, domain: string) => {
   if (!domainExist(tree, domain)) throw new Error('Invalid domain');
```

Comment:
The error should be more descriptive.
ie. "Invalid domain. It does not have the 'type:domain-logic' tag"